### ISSUE TITLE
Fix backwards compatability with older Typescript versions 

### ISSUE THREAD

**Issue by jacobdr** [dcharbonnier/hydrated-ws#130]
Current issues with Typescript 3.0 and 2.9. @dcharbonnier I am happy to make a PR to fix these issues that way TS version should not stop consumers from using this library. How far back do you want me to try to support? 

## TS 3.0

```
./node_modules/ts-typesversions/node_modules/typescript-3.0/bin/tsc

src/pipe/Pipe.ts:102:65 - error TS2345: Argument of type '{ data: string; origin: string; ports: any; source: MessageEventSource; }' is not assignable to parameter of type 'MessageEventInit'.
  Types of property 'source' are incompatible.
    Type 'MessageEventSource' is not assignable to type 'Window'.
      Type 'MessagePort' is not assignable to type 'Window'.
        Property 'Blob' is missing in type 'MessagePort'.

102                 super.dispatchEvent(new MessageEvent("message", {
                                                                    ~
103                     data: e.data.substr(this.prefixLength),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
106                     source: e.source,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
107                 }));
    ~~~~~~~~~~~~~~~~~

src/polyfill/Event.ts:3:7 - error TS2420: Class 'EventPolyfill' incorrectly implements interface 'Event'.
  Property 'deepPath' is missing in type 'EventPolyfill'.

3 class EventPolyfill implements Event {
```

## TS 2.9

```
./node_modules/ts-typesversions/node_modules/typescript-2.9/bin/tsc

src/polyfill/Event.ts:3:7 - error TS2420: Class 'EventPolyfill' incorrectly implements interface 'Event'.
  Property 'deepPath' is missing in type 'EventPolyfill'.

3 class EventPolyfill implements Event {
```

**Comment by dcharbonnier** [comment]
Thanks, I would definitely like to support 2.9 maybe we should just downgrade take for now

**Comment by jacobdr** [comment]
Maybe, but even if we downgrade we'll have to fix the compilation issues. Given how quickly TS 3.0 got bumped to 3.1 I would say that its probably not a good candidate. We could justify moving back to TS 2.9 in the short term. 

I was planning on closing #130 (this ticket) with a hotfix, and then #129 with a more comprehensive solution

**Comment by jacobdr** [comment]
Actually, after we encountered this problem, I started to build a library https://github.com/jacobdr/ts-typesversions to try to systematically address this across the community. I think I might have a POC done by tonight or tomorrow

**Comment by dcharbonnier** [comment]
I released 0.8.2, can you confirm that it works for you so I can close this issue ?

**Comment by jacobdr** [comment]
Yeah man looks good
